packages/transaction-controller/src/TransactionControllerIntegration.test.ts

```diff
@@ -3193,7 +3193,7 @@ describe('TransactionController Integration', () => {
       transactionController.destroy();
     });
 
-    it('should block other attempts to get the nonce lock from the nonceTracker until the first one is released for the given networkClientId', async () => {
+    it('should block attempts to get the nonce lock for the same address from the nonceTracker for the networkClientId until the previous lock is released', async () => {
```

Comment:
```suggestion
    it('should block attempts to get the nonce lock for the same address and chainId combination from a nonceTracker until the lock for a pending nonce for that combination is released', async () => {
```

Reply:
(nit)

Reply:
Oh maybe this is a different (and new test) that I'm proposing ☝️ Where its two different networkClients with the same chainId

Reply:
added https://github.com/MetaMask/core/pull/3869/commits/d72b3f9d8d92b1083d4f71685402815a245b927c